presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -630,6 +631,21 @@ private void addStatsListener(QueryExecution queryExecution)
     {
         Object lock = new Object();
 
+        AtomicBoolean queued = new AtomicBoolean();
+        queryExecution.addStateChangeListener(newValue -> {
```

Comment:
`QUEUED` is the initial state of a query, see the constructor of `QueryStateMachine`, and that state is left after the query moves into planning, running, etc. So, I don't think a state change listener will be called at all with the new state of `QUEUED`.